errutil/error.go

```diff
@@ -40,6 +48,53 @@ func Chain(errs ...error) error {
 	}
 }
 
+// Merge reduces all given errors to a single opaque error
+// instance containing the original information from all
+// errors aggregated but in an opaque way (opposed to Chain).
+//
+// If the len(errs) == 0 it returns nil, if len(errs) == 1 it
+// returns the error itself and it will filter out nil errs.
+//
+// It is specially useful to manage lists of errors and
+// return an error only if any of the errors failed.
+func Merge(errs ...error) error {
+	return Reduce(func(err1, err2 error) error {
+		if err1 == nil && err2 == nil {
+			return nil
+		}
+		if err1 == nil {
+			return err2
+		}
+		if err2 == nil {
+			return err1
+		}
+		return errors.New(err1.Error() + ": " + err2.Error())
+	}, errs...)
+}
+
+// Reduce will reduce all errors to a single one using the
+// provided reduce function.
+//
+// If errs is empty it returns nil, if errs has a single err
+// (len(errs) == 1) it will return the err itself.
+//
+// It won't assume anything else about the given errs, always
+// calling the reduce function, so nil errs on will be passed
+// to the reduce function so it can deal with them.
+//
+// Reduce will panic if the given reduce function panics.
```

Comment:
this is the default for every function, so I don't see why it needs to be documented.
By reading this I'd expect that `Reduce()` deals somehow with panics, like recovering from own panics but throwing user-generated panics.